woocommerce/woocommerce-android
# Issue/9284 after login redirection

Closes: #9284 


### Description


This PR updates the launchMode of MainActivity to ensure a new instance is not started when the activity is already at the top of the stack. 

This PR doesn't implement an optimal fix as it'd be quite complex and is imo not worth the time at the moment. It seems to me the app misuses the`FLAG_ACTIVITY_NEW_TASK` -> at least I don't see any reason why we'd need this flag in our app.

The ideal fix imo would be to remove the usage of `FLAG_ACTIVITY_NEW_TASK` from all the places and rely on the default Android behavior + ensure MainActivity isn't recreated when it's already at the top of the stack.

Either case, this PR only updates the latter part of the code and ensures the MainActivity is not recreated when it's already at the top. This fixes the original issue where the MainActivity used to redirect the user to MyStore screen when it was originally not launched from the OS Launcher but for example from the Login Flow.

More info on peaMlT-64-p2

Let me know if you think this fix is worth it - seems a bit risky, but not too risky, wdyt?

### Testing instructions


- Test Logout
- Test Login
- Test New Order Notification
- Test Shortcuts

1. Login into the app (if you are logged in already, log out and log in)!!
2. Start creating a new product and add for example title
3. Put the app into background
4. Tap on the launcher icon of the app to bring it back to foreground
5. Notice the state of Product Creation is retained
